Prova di Ristrutturazione

Lgegmb opened the pull request:
Accorpamento delle Classi Presentazione e Conferenza in classe Event; ho pensato inoltre di eliminare la classe associativa Loan accorpandola nella classe User

![BozzaRistrutturata](https://user-images.githubusercontent.com/118052439/204802754-540e9626-c869-407a-9658-ba96a9698249.png)

Lgegmb committed:
Prova di Ristrutturazione

Accorpamento delle Classi Presentazione e Conferenza in classe Event; ho pensato inoltre di eliminare la classe associativa Loan accorpandola nella classe User

Lgegmb committed:
Diagramma di prova con consigli prof

Ho eliminato l'enum Type, aggiungendo semplicemente un attributo PublishPeriod:String. Ho aggiunto ISBN per libri e ISSN per collane e riviste; ho messo DOI sugli articoli e sui libri, penserai che per i libri sia ridondante ma ho visto che l'ISBN non varia su ristampe dello stesso libro mentre il DOI è unico e per questo ho pensato fosse meglio avere una chiave candidata del genere. Ho tolto l'associazione leggere e cambiato l'UserName in FCUser perchè ho visto che nelle biblioteche online si usa il CF e la password. Poi mi è nato il pensiero di aggiungere Loan ad User trattando tutti i tipi di lettura come prestito che poi potrebbe rincominciare, solo se il libro è cartaceo ci sta la restituzione e si risolverebbe con un trigger complesso. Fammi sapere cosa ne pensi. Detto questo spero che ti stia divertendo.

lorenzotecchia commented:
Ho pensato un po ai tuoi cambiamenti e a quello che aveva detto il professore, forse forse non ci interessa andare a tracciare chi sta leggendo cosa. Quindi anche le altre relazioni che alla fine sono sempre lettura oppure consultazione secondo me potrebbero esser semplicemente eliminate. Sto esaminando qualche possibile cambiamento. Intanto nella classe libro ci sono due volte gli autori. E' stato fatto di proposito? se si perche?

Lgegmb committed:
ClassDiagramPJ2.png

Lgegmb committed:
Ho pensato anche ad un'altra idea

Autori doppio è un errore di distrazione, ho pensato di vedere tutte le consultazioni come prestiti e la modalità di fruizione fammi sapere cosa ne pensi.

Lgegmb committed:
Merge branch 'lorenzotecchia:main' into main

Lgegmb committed:
Diagramma Ristrutturato
